core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/logical/LogicalInfixExpression.kt

```diff
@@ -2,15 +2,16 @@ package ch.ergon.dope.resolvable.expression.unaliased.type.logical
 
 import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.toBooleanType
+import ch.ergon.dope.resolvable.formatToQueryStringWithBrackets
 import ch.ergon.dope.resolvable.operator.InfixOperator
 import ch.ergon.dope.validtype.BooleanType
 
 class LogicalInfixExpression(
-    left: TypeExpression<BooleanType>,
-    symbol: String,
-    right: TypeExpression<BooleanType>,
+    private val left: TypeExpression<BooleanType>,
+    private val symbol: String,
+    private val right: TypeExpression<BooleanType>,
 ) : TypeExpression<BooleanType>, InfixOperator(left, symbol, right) {
-    override fun toQueryString(): String = toInfixQueryString()
+    override fun toQueryString(): String = formatToQueryStringWithBrackets(left.toQueryString(), symbol, right.toQueryString())
```

Comment:
It's a bit silly, that you inherit from _InfixOperator_, but don't use the defined query-method there. Maybe you can add a parameter toInfixQueryString(useBrackets = true) to the method and call formatToQueryStringWithBrackets from there

Reply:
done